### DISCUSSION TITLE
Force `useSWR` return data fetching not to be undefined with `config.fallbackData`

### DISCUSSION THREAD

**Discussion by tannguyencse19** [vercel/swr#1765] [Q&A]
Continue from #1764 😄 

I want to return static data when `useSWR` fetch error or data is `undefined`.

I try this

```jsx
// <Children />
const { data } = useSWR("/destination", {
  fallbackData: static_data,
});
```

`data` type still `static_data_type | undefined`. Any short way to get around this?

---
Don't know this way **_will also_** make <Children /> useSWR return data not undefined or not?

```jsx

// <Parent />
export const getStaticProps: GetStaticProps = async (
  context
) => {
  const planet = await fetch("/destination");

  return {
    props: {
      SWRFallback: {
        "/destination": planet ? planet : static_data,
      },
    },
  };
};
```

**Comment by iv-stpn** [comment]
+1

It would be great for this behavior to work with SSR as well for dynamic fetching, to have some sort of globally shared variable that could serve as a revalidated cache (e.g. a user object fetched via /api/me in the root layout that is expected to be available throughout the app, and could be mutated once the user updates their profile)
